File: models/user_card.js
import mongoose from 'mongoose';
const { Schema, model } = mongoose;

const playerSchema = new mongoose.Schema({
    card: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Player'
    },
    niveau: {
        type: Number,
        required: true,
        default: 1
    },
    position: {
        type: Number,
        required: true,
        default: -1
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    }
});

export default model('user_card', playerSchema);

File: lib/model/user_card.dart
import 'package:ballerchain/model/user.dart';

class UserCard {
  String? id;
  String? card;
  int? niveau;
  int? position;
  String? user;

  UserCard({
    this.id,
    this.card,
    this.niveau,
    this.position,
    this.user,
  });

  factory UserCard.fromJson(Map<String, dynamic> json) {
    return UserCard(
      id: json['_id'],
      card: json['card'],
      niveau: json['niveau'],
      position: json['position'],
      user: json['user'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'card': card,
      'niveau': niveau,
      'position': position,
      'user': user,
    };
  }
}
